client/html/src/Client/Html/Catalog/Product/Standard.php

```diff
@@ -0,0 +1,468 @@
+		 * domains (attribute, media, price, product, text, etc. are domains)
+		 * whose items are fetched from the storage. Please keep in mind that
+		 * the more domains you add to the configuration, the more time is required
+		 * for fetching the content!
+		 *
+		 * This configuration option can be overwritten by the "client/html/catalog/product/domains"
+		 * configuration option that allows to configure the domain names of the
+		 * items fetched specifically for all types of product listings.
+		 *
+		 * @param array List of domain names
+		 * @since 2014.03
+		 * @category Developer
+		 * @see client/html/catalog/product/domains
+		 */
+		$domains = $config->get( 'client/html/catalog/domains', ['media', 'price', 'text'] );
+
+		/** client/html/catalog/product/domains
+		 * A list of domain names whose items should be available in the catalog product view template
+		 *
+		 * The templates rendering product lists usually add the images, prices
+		 * and texts associated to each product item. If you want to display additional
+		 * content like the product attributes, you can configure your own list of
+		 * domains (attribute, media, price, product, text, etc. are domains)
+		 * whose items are fetched from the storage. Please keep in mind that
+		 * the more domains you add to the configuration, the more time is required
+		 * for fetching the content!
+		 *
+		 * This configuration option overwrites the "client/html/catalog/domains"
+		 * option that allows to configure the domain names of the items fetched
+		 * for all catalog related data.
+		 *
+		 * @param array List of domain names
+		 * @since 2019.06
+		 * @category Developer
+		 * @see client/html/catalog/domains
+		 * @see client/html/catalog/detail/domains
+		 * @see client/html/catalog/stage/domains
+		 * @see client/html/catalog/lists/domains
+		 */
+		$domains = $config->get( 'client/html/catalog/product/domains', $domains );
+
+		/** client/html/catalog/product/product-codes
+		 * List of codes of products to load for the current list.
+		 * Should be set dynamically through some integration plugin,
+		 * to allow a list of products with configurable products.
+		 *
+		 * @param string List of codes of products to load for the current list
+		 * @since 2019.06
+		 * @category Developer
+		 */
+		$productCodes = $config->get( 'client/html/catalog/product/product-codes', [] );
+
+		$products = \Aimeos\Controller\Frontend::create( $context, 'product' )
+			->compare( '==', 'product.code', $productCodes )
+			->slice( 0, count( $productCodes ) )
+			->uses( $domains )
+			->search();
+
+		// Sort products by the order given in the configuration "client/html/catalog/product/product-codes".
+		$productCodesOrder = array_flip( $productCodes );
+		usort( $products, function ( $a, $b ) use ( $productCodesOrder ) {
+			return $productCodesOrder[$a->getCode()] - $productCodesOrder[$b->getCode()];
+		} );
+
+
+		if ( $config->get( 'client/html/catalog/product/basket-add', false ) ) {
```

Comment:
This isn't necessary any more because at least since 2019.01, the product item contains all referenced items including their references